### PR TITLE
Gtl table

### PR DESCRIPTION
**Pull request by karelhala**
New component with name miqDataTable `miq-data-table` was added.
* [x] update docs
* [x] tests for controller
* [x] tests for component

selection_074

### CODE REVIEW

**Comment by karelhala** [comment]
TODO: when switching pages selected items are forgotten, this is more relevant to main MiQ repo, but keeping

**Comment by karelhala** [comment]
@martinpovolny @himdel @mtho11 all tests are fine so this thing is good to go.

File: src/gtl/components/data-table/data-table.html [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    </tr>
+    </tbody>
+  </table>
+  <div class="dataTables_footer">
+    <div class="dataTables_paginate paging_bootstrap_input">
+      <ul class="pagination">
+        <li ng-class="{disabled: tableCtrl.currentPageView === 1}"
+            class="first"
+            ng-click="tableCtrl.setPage(1)">
+          <span class="i fa fa-angle-double-left"></span>
```
Comment (himdel) [inline src/gtl/components/data-table/data-table.html]: Is `.i` a thing? Not seeing any class of that name in patternfly or here..

Comment (karelhala) [inline src/gtl/components/data-table/data-table.html]: Yeah, I copied this from old implementation of table with paging, that's why it's there. But I'll remove it (not really necessary there).

**Comment by karelhala** [comment]
ping @mtho11 @martinpovolny @himdel

**Comment by martinpovolny** [comment]
Ok, merging this. It's not used yet so any further feedback can be addressed in a follow-up PR.

**Comment by martinpovolny** [comment]
pagination allows me to go to negative page number

**Pull request merged**